pyproject.toml

```
@@ -37,8 +37,9 @@ dev = [
     "mypy",  # linting
     "myst-parser",  # markdown parsing
     "pytest",  # testing
-    "ruff",  # linting
-    "sphinx"  # documentation
+    "ruff==0.0.292",  # linting
+    "sphinx",  # documentation
+    "pre-commit==3.4.0", # managages/maintains pre-commit hooks
```

```suggestion
    "pre-commit==3.4.0", # manages and maintains pre-commit hooks
```

Just a little typo fix. I think you can click a button to accept it

Your careful attention to adding comments @VivekKumarKamal has been nice. You're doing great work.